client/src/main/java/com/paypal/selion/configuration/ListenerInfo.java

```diff
@@ -33,24 +38,62 @@
      *            - The JVM argument that should be read to decide if the listener is to be enabled/disabled.
      */
     public ListenerInfo(Class<?> className, String jvmArgToParse) {
+        this(className, jvmArgToParse, ENABLE_LISTENER_BY_DEFAULT);
+    }
+
+    /**
+     * Saves information about listener class which is enabled/disabled per its VM argument. When VM argument is
+     * unavailable, it's enabled/disabled is decided by the caller via (@code isEnabledWhenJvmAbsent}.
+     * 
+     * @param className
+     *            - The Class name.
+     * @param jvmArgToParse
+     *            - The JVM argument that should be read to decide if the listener is to be enabled/disabled.
+     * @param isListenerEnabledWhenJvmAbsent
+     *            - The default boolean for whether enabled/disabled when the JVM argument is not found.
+     */
+    public ListenerInfo(Class<?> className, String jvmArgToParse, boolean isListenerEnabledWhenJvmAbsent) {
         this.listenerClassName = className.getCanonicalName();
-        this.enabled = getBooleanValFromVMArg(jvmArgToParse);
+        this.listenerEnabled = getBooleanValFromVMArg(jvmArgToParse, isListenerEnabledWhenJvmAbsent);
     }
 
+    /**
+     * Gets the class name of the listener.
+     */
     public String getListenerClassName() {
         return listenerClassName;
     }
 
     /**
+     * Gets a boolean indicating whether or not the listener is enabled.
+     * 
      * @return - <code>true</code> if the listener is to be enabled, <code>false</code> otherwise.
      */
     public boolean isEnabled() {
-        return enabled;
+        return this.listenerEnabled;
     }
 
+    /**
+     * Returns boolean value of the JVM argument when available, else returns the system default.
+     * 
+     * @param vmArgValue
+     *          
+     */
     static boolean getBooleanValFromVMArg(String vmArgValue) {
-        boolean flag = true;
+        return getBooleanValFromVMArg(vmArgValue, ENABLE_LISTENER_BY_DEFAULT);
+    }
+
+    /**
+     * Returns boolean value of the JVM argument when available, else returns the {@code isEnabledWhenJvmAbsent}.
+     * 
+     * @param vmArgValue
+     *      The VM argument name.
+     * @param isListenerEnabledWhenJvmAbsent
```

Comment:
How can the JVM be absent in a java process? I suspect you mean to call this more like `defaultStateWhenNotDefined`